contracts/test/integration/MultiSubnet.t.sol

```diff
@@ -308,6 +334,110 @@ contract MultiSubnetTest is Test, IntegrationTestBase {
         assertEq(recipient.balance, amount);
     }
 
+    function testMultiSubnet_DeflationaryErc20_ReleaseFromChildToParent() public {
+        address caller = address(new MockIpcContract());
+        address recipient = address(new MockIpcContractPayable());
+        uint256 amount = 4098;
+
+        deflationaryToken.transfer(caller, amount);
+        vm.prank(caller);
+        deflationaryToken.approve(rootSubnet.gatewayAddr, amount);
+
+        vm.deal(deflationaryTokenSubnet.subnetActorAddr, DEFAULT_COLLATERAL_AMOUNT);
+        vm.deal(caller, 1 ether);
+
+        vm.prank(deflationaryTokenSubnet.subnetActorAddr);
+        registerSubnetGW(DEFAULT_COLLATERAL_AMOUNT, deflationaryTokenSubnet.subnetActorAddr, rootSubnet.gateway);
+
+        vm.prank(caller);
+        vm.expectRevert(); // half of tokens burned on transfer
+        rootSubnet.gateway.manager().fundWithToken(
+            deflationaryTokenSubnet.id,
+            FvmAddressHelper.from(address(caller)),
+            amount
+        );
+
+        vm.prank(caller);
+        rootSubnet.gateway.manager().fundWithToken(
```

Comment:
Shouldn't we assert that: (a) the circ supply matches the deflated amount, and (b) the gateway's token balance matches the deflated amount. Same for the inflationary and nil test cases.

Reply:
@snissn ^^

Reply:
Thanks, that makes sense, adding